app_dart/lib/src/service/github_service.dart

```diff
@@ -15,27 +15,40 @@ class GithubService {
   final GitHub github;
 
   /// Lists the commits of the provided repository [slug] and [branch].
-  Future<List<RepositoryCommit>> listCommits(
-      RepositorySlug slug, String branch, int hours, Config config) async {
-    final Duration time = Duration(hours: hours);
+  Future<List<RepositoryCommit>> listCommits(RepositorySlug slug, String branch,
+      int lastCommitTimestampMills, Config config) async {
     ArgumentError.checkNotNull(slug);
     final PaginationHelper paginationHelper = PaginationHelper(github);
 
+    /// Return only one page when this is a new branch. Otherwise it will
+    /// return all commits prior to this release branch commit, leading to
+    /// heavy workload.
+    int pages;
+    if (lastCommitTimestampMills == 0) {
```

Comment:
What happens if this check fails? should we add a default for pages?

Reply:
It will return all results. No need for a default value: https://github.com/SpinlockLabs/github.dart/blob/master/lib/src/common/util/pagination.dart#L54

The `null` has been taken care of.